.github/workflows/build.yml

```diff
@@ -1,11 +1,24 @@
 name: ci
 on:
+  push:
   pull_request:
-    paths-ignore:
-    - '**/*.md'
 jobs:
+  skip_check:
+    runs-on: ubuntu-latest
+    outputs:
+      should_skip: ${{ steps.skip_check.outputs.should_skip }}
+    steps:
+      - id: skip_check
+        uses: fkirc/skip-duplicate-actions@v3.4.0
```

Comment:
There's alternative actions we can use but I wasn't a huge fan of their ignore filtering behaviors when playing around with them locally. Another viable alternative would be to perform this filtering ourselves through the REST API using the /pulls/<PR #>/files endpoint.

Reply:
There's likely a couple of more alternatives the more I think about:

- Run CI on every PR, regardless of content. This is the most costly option but most of the CI runs we like to skip are markdown-only PRs and most of the OLM documentation is housed in another repository.
- Dynamically determine required checks; this would require removing required status checks entirely and updating our tide configuration to accommodate that requirement
- Dynamically determine when to skip certain required checks (which is what this PR attempts to do)
  - Use the rest API and parse through the list of files that were changed in the PR
  - Use a custom action to perform this filtering as a job that runs before any CI-related jobs (e.g. a prerequisite job)
  - Could also play around with a workflow_dispatch singleton that triggers CI-related jobs but we'd likely run into the same issue of filtering + existing required status checks blocking PRs